utility/synapse-utilities-common/src/main/java/io/americanexpress/synapse/utilities/common/validator/AnyOfValidator.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.americanexpress.synapse.utilities.common.validator;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.apache.commons.beanutils.PropertyUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import org.apache.commons.lang3.StringUtils;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+/**
+ * {@code AnyOfValidator} validates that at least one of the fields is provided.
+ *
+ * @author sahilzmaharjan
+ */
+public class AnyOfValidator implements ConstraintValidator<AnyOf, Object> {
+
+    /**
+     * Fields to be validated.
+     */
+    private String[] fieldNames;
+
+    /**
+     * Initialize the validator in preparation for isValid calls.
+     *
+     * @param constraintAnnotation annotation instance for a given constraint declaration.
+     */
+    @Override
+    public void initialize(AnyOf constraintAnnotation) {
+        this.fieldNames = constraintAnnotation.fieldNames();
+    }
+
+    /**
+     * Check if at least one of the fields is provided.
+     *
+     * @param object object to validate.
+     * @param constraintValidatorContext context in which constraint is evaluated.
+     * @return true if any one of the fields is provided.
+     */
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+        if(object == null || ArrayUtils.isEmpty(fieldNames)) {
```

Comment:
```suggestion
        if(object == null || ArrayUtils.isEmpty(fieldNames)  || fieldNames.length < 1) {
```
did you say yesterday to add this check as well?

Reply:
Had added the last check, but realized ArrayUtils.isEmpty is checking the same thing.